src/main/scala/akka/contrib/process/BlockingProcess.scala

```diff
@@ -97,77 +97,55 @@ class BlockingProcess(
   override val supervisorStrategy: SupervisorStrategy =
     SupervisorStrategy.stoppingStrategy
 
-  private val process = startProcess()
-
-  // This shutdown hook is required because the stdio actors will be blocked on a read. The only
-  // reliable means of unblocking them is to destroy the thing they're listening to.
-  context.actorOf(ShutdownHook.props(() => process.destroy()), "process-destroyer")
+  override def preStart(): Unit = {
+    val process: JavaProcess = {
+      import JavaConverters._
+      val preparedCommand = if (Helpers.isWindows) command map winQuote else command
+      val pb = new JavaProcessBuilder(preparedCommand.asJava)
+      pb.environment().putAll(environment.asJava)
+      pb.directory(directory)
+      pb.start()
+    }
 
-  private val stdin =
-    context.actorOf(OutputStreamSubscriber.props(process.getOutputStream), "stdin")
+    val destroyer =
```

Comment:
That's a good point, I've made it more resilient by putting the instantiation of the destroyer in a finally-block.